Correct Scalable Sizing Crash for Various Zone Equipment

RKStrand opened the pull request:
## Pull request overview

This pull request corrects a crash that happened with the scalable sizing routines.  At issue was a variable was not being reset to false consistently across all of the different equipment that use scalable sizing.  This was causing a crash when scalable sizing was being used for a high temperature radiant heater and then a DX coil was being sized.  The unexpected wrong value for a critical flag caused the wrong code to be executed and as a result a hard crash.  This was corrected for the high temperature radiant heater as well as other equipment which use this same flag.  The new code corrects the problem and the user defect input file now runs as it should.  This problem was eliminated for other equipment combinations as well.  The original problem was logged as GitHub Issue #5728.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ x ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [ x ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue #5728.
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes n/a

RKStrand committed:
Fix of crash

This fix avoids the crash that discovered when a high temperature
radiant heater is autosized using scalable sizing factors as in the
defect example idf.  This needs to be looked into for other equipment
that uses the scalable sizing process to make sure that this is not an
issue elsewhere.  Committing this simple fix to trigger the CI runs to
make sure it doesn’t introduce a new problem.

RKStrand committed:
Correct other missing assignments

DataScalableCapSizingOn needs to be consistent in how it is assigned
across a variety of pieces of equipment that fit in the zone equipment
category.  These fixes make things consistent and should avoid the
crash that was noted with the high temperature radiant system.

RKStrand committed:
Merge branch 'develop' into 129956421-Scalable-Sizing-HTRad-Heat-Crash

RKStrand committed:
Unit test verifying correct assignment of flag

This new unit test verifies that the DataScalableSizingON flag is rest
properly.  It was being left to true and this was causing a crash under
certain equipment combinations.  The original input file that caused
the problem combined high temperature radiant heaters with DX coils.
Now, all zone type equipment has this flag set back to false once the
sizing process is complete.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 129956421-Scalable-Sizing-HTRad-Heat-Crash

mjwitte commented:
@rraustad Mr. Sizing Guru, do these changes look ok to you?

mjwitte commented on tst/EnergyPlus/unit/HighTempRadiantSystem.unit.cc:
```diff
@@ -137,5 +141,35 @@ namespace EnergyPlus {
+		int RadSysNum;
+		int SizingTypesNum;
+		
+		HighTempRadiantSystem::clear_state();
```
This doesn't need to be here.  I'll remove it.  It gets called from the fixture.

RKStrand replied:
@mjwitte I'll look at these and add as appropriate (DataScalableCapSizingON = false).

RKStrand replied:
@mjwitte And thanks for removing the extraneous clear_state.  Still not 100% on my understanding of those, as you can tell.

mjwitte committed:
HighTempRadiantSystem.unit.cc remove extraneous clear_state

mjwitte reviewed (commented):
@RKStrand Based on a search of the code for `DataScalableCapSizingOn`, it looks like HVACVariableRefrigerantFlow.cc, PackagedTerminalHeatPump.cc, and WindowAC.cc are missing trailing `DataScalableCapSizingON = false` statements.

RKStrand committed:
Addition of more flag assignments

Brought three other scalable autosized models up to the same level as
all of the other ones that use the flag DataScalableCapSizingON.  This
flag should now avoid the crash that precipitated this defect in all
models that use this variable.

RKStrand commented:
@mjwitte I added the proper assignment of the flag to the other models that use the DataScalableCapSizingON flag.  So, I think that addresses your comments, Mike.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 129956421-Scalable-Sizing-HTRad-Heat-Crash

rraustad commented:
Just got back from vacation. These changes look correct to me.
